src/useScrollSync.ts

```diff
@@ -0,0 +1,91 @@
+import type React from "react";
+import { useEffect } from "react";
+
+interface Options {
+  proportional?: boolean;
```

Comment:
Maybe also an enum? “proportional” | “absolute”?

Reply:
Something like `mode: 'proportional' | 'absolute'`?

Does it make sense to default this to `'proportional'` or `'absolute'`? I know the current implementation defaults to `proportional=true`, but I feel like `'absolute'` might be a more natural default.

Actually, I might be most in favor of not defaulting this to anything at all and requiring user to specify.

Reply:
Yes, exactly. I think defaulting to `proportional` is more appropriate but I'm open for making it a required prop.

Reply:
Made `mode` a required parameter for now.